compiler/rustc_arena/src/lib.rs

```diff
@@ -197,23 +197,24 @@ impl<T> TypedArena<T> {
         start_ptr
     }
 
+    /// Allocates the elements of this iterator into a contiguous slice in the `TypedArena`.
+    ///
+    /// Note: this may not have the memory characteristics you expect. For reasons of reentrancy and
```

Comment:
It's unclear to me what "memory characteristics" means, exactly. Maybe just remove that sentence and start with "Note: for reasons...".